packages/editor/src/components/combo-box/combo-box.ts

```diff
@@ -0,0 +1,173 @@
+import { css, html, LitElement, type PropertyValues } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import type { ComboBoxOption } from './types.ts';
+
+@customElement('clippy-combo-box')
+export class ComboBox extends LitElement {
+  @property({ type: Boolean }) disabled = false;
+  @property({ type: Boolean }) readOnly = false;
+  @property({ type: Number }) private activeIndex = -1;
+  @property({ type: Boolean }) private isOpen = false;
+  @property({ type: Array }) private options: ComboBoxOption[] = [];
+  @property({ type: String }) value = '';
+  @property({ type: Function }) onSelect = (value: string) => value;
+
+  override willUpdate(changedProperties: PropertyValues) {
+    if (changedProperties.has('options')) {
+      const activeOption = this.options.find((option) => option.active);
+      if (activeOption) {
+        this.value = activeOption.label;
+      }
+    }
+  }
+
+  static override readonly styles = [
+    css`
+      :host {
+        position: relative;
+        display: inline-block;
+      }
+      .clippy-combo-box {
+        inline-size: 100%;
+        max-inline-size: 500px;
+        position: relative;
+      }
+      .clippy-combobox__input {
+        line-height: 1.8em;
+        background-color: #fff;
+        border: 1px solid var(--basis-color-action-1-color-active);
+        border-radius: 3px;
+        color: var(--basis-color-action-1-color-active);
+        padding-block: var(--basis-space-inline-sm);
+        padding-inline: var(--basis-space-inline-md);
+      }
+    `,
+  ];
+
+  private handleInput(e: Event) {
```

Comment:
t is iets veiliger om specifiek bij event handlers arrow functions te schrijven zodat je als je ze zou gebruiken icm `addEventListener` je:
1) een `this` hebt die verwijst naar het component ipv de functie,
2) je m kan verwijderen met een `removeEventListener`. 

Een `.bind(this)` lijkt misschien te helpen bij punt 1, maar bij punt 2 zal de `bind(this)` een nieuwe referentie maken en dus nooit de remove kunnen doen.